src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-list/pool-list.component.spec.ts

```diff
@@ -267,4 +273,42 @@ describe('PoolListComponent', () => {
       expect(component.getPoolDetails(pool)).toEqual(expected);
     });
   });
+
+  describe('getSelectionTiers', () => {
+    let pools: Pool[];
+    const setSelectionTiers = (tiers: number[]) => {
+      component.selection.selected = [
+        {
+          tiers
+        }
+      ];
+      component.selection.update();
+      component.getSelectionTiers();
+    };
+
+    beforeEach(() => {
+      pools = [];
+      setUpPools(pools);
+    });
+
+    it('should select multiple existing cache tiers', () => {
+      setSelectionTiers([0, 1, 2]);
+      expect(component.selectionCacheTiers).toEqual(pools);
+    });
+
+    it('should select correct existing cache tier', () => {
+      setSelectionTiers([0]);
+      expect(component.selectionCacheTiers).toEqual([{ pg_num: 256, pool: 0, pool_name: 'a' }]);
+    });
+
+    it('should not select cache tier if id is invalid', () => {
+      setSelectionTiers([-1]);
+      expect(component.selectionCacheTiers).toEqual([]);
+    });
+
+    it('should not select cache tier if empty', () => {
+      setSelectionTiers([]);
+      expect(component.selectionCacheTiers).toEqual([]);
+    });
```

Comment:
Could you please include the test that failed with the previous solution? This way you will make sure that the code works as expected also in the future.

```diff
+    // This test failed with the previous solution
+    it('should be able to selected one pool with multiple tiers, than with a single tier, than with no tiers', () => {
+      setSelectionTiers([0, 1, 2]);
+      expect(component.selectionCacheTiers).toEqual(pools);
+      setSelectionTiers([0]);
+      expect(component.selectionCacheTiers).toEqual([{ pg_num: 256, pool: 0, pool_name: 'a' }]);
+      setSelectionTiers([]);
+      expect(component.selectionCacheTiers).toEqual([]);
+    });
```